### PR TITLE
Introduce GKE Integration job installing Compass on top of Kyma

### PR DESCRIPTION
**Pull request by pkosiec**
**Description**

Changes proposed in this pull request:

- Introduce GKE Integration job installing Compass on top of Kyma

Currently this job is optional and eventually it will replace existing `kyma-gke-compass-integration` on Kyma repository.

**Related issue(s)**

Proposal: https://github.com/kyma-incubator/compass/blob/master/docs/internal/proposals/separate-compass/separate-compass.md

### CODE REVIEW

File: templates/templates/kyma-gke-separate-compass-integration.yaml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  preset-gc-compute-envs: "true"
+  preset-gc-project-env: "true"
+  preset-docker-push-repository-gke-integration: "true"
+  preset-dind-enabled: "true"
+  preset-kyma-artifacts-bucket: "true"
+  preset-gardener-azure-kyma-integration: "true"
+
+presubmits: # runs on PRs
+  kyma-project/kyma:
+    - name: pre-master-kyma-gke-separate-compass-integration
```
Comment (tgorgol) [inline templates/templates/kyma-gke-separate-compass-integration.yaml:61]: Let's set those jobs to optional for now.

File: templates/templates/kyma-gke-separate-compass-integration.yaml [hunk 2]
```diff
@@ -0,0 +1,107 @@
+test_infra_ref: &test_infra_ref
+  org: kyma-project
+  repo: test-infra
+  path_alias: github.com/kyma-project/test-infra
+
+kyma_ref: &kyma_ref
```
Comment (tgorgol) [inline templates/templates/kyma-gke-separate-compass-integration.yaml:6]: This ref seems unused, I think we can get rid of it.

File: prow/scripts/cluster-integration/kyma-gke-separate-compass-integration.sh [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+export GCLOUD_PROJECT_NAME="${CLOUDSDK_CORE_PROJECT}"
+export GCLOUD_COMPUTE_ZONE="${CLOUDSDK_COMPUTE_ZONE}"
+
+#Local variables
+DNS_SUBDOMAIN="${COMMON_NAME}"
+KYMA_SCRIPTS_DIR="${KYMA_SOURCES_DIR}/installation/scripts"
+KYMA_RESOURCES_DIR="${KYMA_SOURCES_DIR}/installation/resources"
+
+INSTALLER_YAML="${KYMA_RESOURCES_DIR}/installer.yaml"
+INSTALLER_CR="${KYMA_RESOURCES_DIR}/installer-cr-compass-dependencies.yaml.tpl"
```
Comment (tgorgol) [inline prow/scripts/cluster-integration/kyma-gke-separate-compass-integration.sh:79]: Shouldn't we be using full kyma installation here (with runtime agent etc.)?

File: prow/scripts/cluster-integration/kyma-gke-separate-compass-integration.sh [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+  kubectl create namespace "compass-installer"
+  applyCommonOverrides "compass-installer"
+  applyCompassOverrides
+
+  echo "Use master Compass artifacts"
+  readonly COMPASS_MASTER_ARTIFACTS="${COMPASS_DEVELOPMENT_ARTIFACTS_BUCKET}/master"
+  readonly COMPASS_TMP_DIR="/tmp/compass-gke-integration"
+  gsutil cp "${COMPASS_MASTER_ARTIFACTS}/compass-installer.yaml" ${COMPASS_TMP_DIR}/compass-installer.yaml
+  gsutil cp "${COMPASS_MASTER_ARTIFACTS}/is-installed.sh" ${COMPASS_TMP_DIR}/is-installed.sh
+	"${KYMA_SCRIPTS_DIR}"/concat-yamls.sh ${COMPASS_TMP_DIR}/compass-installer.yaml "${INSTALLER_CR}" \
```
Comment (tgorgol) [inline prow/scripts/cluster-integration/kyma-gke-separate-compass-integration.sh:338]: There's no need to concat component list with this artifact as it already contains one, also this `INSTALLER_CR` is for Kyma, not Compass installation.

File: prow/scripts/cluster-integration/kyma-gke-separate-compass-integration.sh [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+	| kubectl apply -f ${COMPASS_TMP_DIR}/compass-installer.yaml
```
Comment (tgorgol) [inline prow/scripts/cluster-integration/kyma-gke-separate-compass-integration.sh:339]: I think this pipe should be removed, right?

File: prow/scripts/cluster-integration/kyma-gke-separate-compass-integration.sh (hunk 3 above)
Comment (pkosiec) [reply prow/scripts/cluster-integration/kyma-gke-separate-compass-integration.sh:79]: Yes, this is the file - I agree, I should've named it better. Fixed

**Pull request merged**